packages/apps/testbench/src/components/ItemList.tsx

```diff
@@ -0,0 +1,76 @@
+//
+// Copyright 2024 DXOS.org
+//
+
+import { X } from '@phosphor-icons/react';
+import React from 'react';
+
+import * as E from '@dxos/echo-schema';
+import { Button, Input } from '@dxos/react-ui';
+import { automerge, createBasicExtensions, useDocAccessor, useTextEditor } from '@dxos/react-ui-editor';
+
+import { type ItemType } from '../data';
+
+// TODO(burdon): List and Table variants.
+
+export type ItemListProps<T> = {
+  objects: T[];
+} & Pick<ItemProps<T>, 'onDelete'>;
+
+export const ItemList = ({ objects, ...props }: ItemListProps<ItemType>) => {
+  return (
+    <div className='flex flex-col grow overflow-hidden'>
+      <div className='flex flex-col overflow-y-scroll pr-2'>
+        {objects.map((object) => (
+          <Item key={object.id} object={object} {...props} />
+        ))}
+      </div>
+    </div>
+  );
+};
+
+export type ItemProps<T> = {
+  object: T;
+  onDelete: (id: string) => void;
+};
+
+// TODO(burdon): Use ui list with key nav/selection.
+// TODO(burdon): Toggle options to show deleted.
+export const Item = ({ object, onDelete }: ItemProps<ItemType>) => {
+  // const { themeMode } = useThemeContext(); // TODO(burdon): Set-up theme.
+  const { doc, accessor } = useDocAccessor(object.text!); // TODO(burdon): [API]: Accept undefined?
```

Comment:
note: `useDocAccessor` is going away https://github.com/dxos/dxos/pull/6218

Reply:
I've re-added it back since it still seems to be used in a couple places. Let's discuss what kind of API we want there